src/main/java/duke/command/DeadlineCommand.java

```diff
@@ -0,0 +1,46 @@
+package duke.command;
+
+import duke.task.Deadline;
+import duke.task.Task;
+import duke.util.*;
+
+import java.time.LocalDateTime;
+import java.util.Map;
+
+
+/**
+ * The DeadlineCommand class encapsulates the run method for the deadline command.
+ */
+public class DeadlineCommand implements DukeActions {
+
+    /**
+     * Performs the actions for the Deadline Command when activated
+     * @param map The parsed command
+     * @param list The tasklist
+     * @param database The database to write to
+     * @param config The configuration settings
+     * @param ui The UI object to interact with
+     * @return boolean to indicate the end of the listen operation
+     * @throws DukeException When erroneous inputs are given.
+     */
+    @Override
+    public boolean runAndCanContinue(Map<String, String> map, DukeTaskList list, DukeDB database, DukeConfig config, Ui ui)
```

Comment:
Does the naming sound boolean-like? maybe try: `isCanRunAndContinue`

Reply:
the boolean is however not asserting if it can run. it is however checking if we should continue running the bot